src/execution/service.ts

```diff
@@ -776,6 +803,22 @@ export class ServiceScriptExecution extends BaseExecutionWithCommand<ServiceScri
   private _onChildStarted() {
     switch (this._state.id) {
       case 'starting': {
+        if (this._state.readyMonitor !== undefined) {
+          this._state = {
+            id: 'readying',
+            child: this._state.child,
+            fingerprint: this._state.fingerprint,
+            started: this._state.started,
+            readyMonitor: this._state.readyMonitor,
```

Comment:
What you have written is more precise. But 🏌️:

`this.state = { ...this._state, id: 'readying' };`

Reply:
I guess what I don't like about that as a general pattern for state transitions is that it will also copy properties from the old state that are not declared in the new state, without a type error.

The worst case would be something like:
1. State A has required property `foo`
2. State B has no property `foo`, but we copy it with the `...`
3. State C has *optional* property `foo`, which *should* be undefined, but we inadvertently propagate the `foo` from A to C using the `...`

(That situation isn't a problem in practice currently, but that's why I prefer just being explicit)

Reply:
Agree!